llvm/lib/Target/AIE/AIEEliminateDuplicatePHI.cpp

```diff
@@ -120,26 +120,26 @@ class AIEEliminateDuplicatePHI : public MachineFunctionPass {
   bool processMBB(MachineBasicBlock &MBB, MachineRegisterInfo &MRI,
                   MachineIRBuilder &MIB, GISelObserverWrapper &Observer) {
     bool Changed = false;
-    auto LastPHI = MBB.getFirstNonPHI();
-    for (auto &MI : make_early_inc_range(MBB.phis())) {
-      if (!isPointerTypePHI(MRI, MI))
-        continue;
-
-      // Ensure that next MI is valid
-      auto NextMI = std::next(MachineBasicBlock::instr_iterator(MI));
-      if (NextMI == MBB.instr_end() || NextMI == LastPHI)
+    const auto End = MBB.end();
+    for (auto PhiItA = MBB.begin(); PhiItA != End; ++PhiItA) {
+      auto &PhiA = *PhiItA;
+      if (PhiA.getOpcode() != TargetOpcode::G_PHI)
         break;
-
-      for (auto &PHI : make_early_inc_range(make_range(
-               NextMI, MachineBasicBlock::instr_iterator(*LastPHI)))) {
```

Comment:
The error was here. LastPHI was at .end(), which means you cannot dereference it to create an instr_iterator. Note that in that case the validity check reads `if (NextMI == end || NextMI == end)`